frontend/packages/console-app/src/components/quick-starts/QuickStartController.tsx

```diff
@@ -39,41 +52,73 @@ const QuickStartController: React.FC<QuickStartControllerProps> = ({ quickStart
   );
 
   const handleNext = React.useCallback(() => {
+    if (status === QuickStartStatus.NOT_STARTED) startQuickStart();
     if (status === QuickStartStatus.COMPLETE && taskNumber === totalTasks)
       return handleQuickStartChange('');
 
-    return nextStep(totalTasks);
-  }, [handleQuickStartChange, nextStep, status, taskNumber, totalTasks]);
+    if (
+      status === QuickStartStatus.IN_PROGRESS &&
+      taskStatus !== QuickStartTaskStatus.INIT &&
+      taskNumber === totalTasks - 1
+    )
+      completeQuickStart();
+
+    if (taskStatus === QuickStartTaskStatus.INIT)
+      return handleTaskStatusChange(QuickStartTaskStatus.REVIEW);
+
+    if (taskNumber < totalTasks) return setQuickStartTaskNumber(name, taskNumber + 1);
+
+    return null;
+  }, [
+    completeQuickStart,
+    handleQuickStartChange,
+    handleTaskStatusChange,
+    name,
+    setQuickStartTaskNumber,
+    startQuickStart,
+    status,
+    taskNumber,
+    taskStatus,
+    totalTasks,
+  ]);
 
   const handleBack = React.useCallback(() => {
-    return previousStep();
-  }, [previousStep]);
+    if (taskNumber > -1) return setQuickStartTaskNumber(name, taskNumber - 1);
+
+    return null;
+  }, [name, setQuickStartTaskNumber, taskNumber]);
 
   const handleTaskSelect = React.useCallback(
     (selectedTaskNumber: number) => {
       setQuickStartTaskNumber(name, selectedTaskNumber);
+      startQuickStart();
     },
-    [name, setQuickStartTaskNumber],
+    [name, setQuickStartTaskNumber, startQuickStart],
   );
 
+  const headerHeight = panelHeadReference.current?.offsetHeight;
+  const computedMaxHeight = window.innerHeight - (headerHeight + 6 * 24); // CSS equivalent of calc(viewportheight - (headerheight + 6 * spacer-md))
```

Comment:
Do not compute the height based on window height. Always create components to fit their available height without assuming you know the parent container and its size.

A flow layout that uses sticky or flex or grid would work best.

Reply:
converted to flex